crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -70,139 +73,106 @@ async fn sys_validation_workflow_inner(
-        }
-    });
-
-    // Create a stream that will chunk up to NUM_CONCURRENT_OPS * 100 ready results.
-    let mut iter =
-        tokio_stream::wrappers::ReceiverStream::new(rx).ready_chunks(NUM_CONCURRENT_OPS * 100);
-
-    let mut total = 0;
-    let mut round_time = start.is_some().then(std::time::Instant::now);
-    // Pull in a chunk of results.
-    while let Some(chunk) = iter.next().await {
-        let num_ops: usize = chunk.iter().map(|c| c.len()).sum();
-        tracing::debug!("Committing {} ops", num_ops);
-        let (t, a, m, r) = space
-            .dht_db
-            .write_async(move |txn| {
-                let mut total = 0;
-                let mut awaiting = 0;
-                let mut missing = 0;
-                let mut rejected = 0;
-                for outcome in chunk.into_iter().flatten() {
-                    let (op_hash, outcome, dependency) = outcome?;
-                    match outcome {
-                        Outcome::Accepted => {
-                            total += 1;
-                            put_validation_limbo(
-                                txn,
-                                &op_hash,
-                                ValidationLimboStatus::SysValidated,
-                            )?;
-                        }
-                        Outcome::AwaitingOpDep(missing_dep) => {
-                            awaiting += 1;
-                            // TODO: Try and get this dependency to add to limbo
-                            //
-                            // I actually can't see how we can do this because there's no
-                            // way to get an DhtOpHash without either having the op or the full
-                            // action. We have neither that's why where here.
-                            //
-                            // We need to be holding the dependency because
-                            // we were meant to get a StoreRecord or StoreEntry or
-                            // RegisterAgentActivity or RegisterAddLink.
-                            let status = ValidationLimboStatus::AwaitingSysDeps(missing_dep);
-                            put_validation_limbo(txn, &op_hash, status)?;
-                        }
-                        Outcome::MissingDhtDep => {
-                            missing += 1;
-                            // TODO: Not sure what missing dht dep is. Check if we need this.
-                            put_validation_limbo(txn, &op_hash, ValidationLimboStatus::Pending)?;
-                        }
-                        Outcome::Rejected => {
-                            rejected += 1;
-                            if let Dependency::Null = dependency {
-                                put_integrated(txn, &op_hash, ValidationStatus::Rejected)?;
-                            } else {
-                                put_integration_limbo(txn, &op_hash, ValidationStatus::Rejected)?;
+        },
+        |batch| {
```

Comment:
At a glance it seems possible to move this function into the batch file. I'd prefer that.